Tabloid/Repositories/TagRepository.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Configuration;
+using Tabloid.Models;
+
+namespace Tabloid.Repositories
+{
+    public class TagRepository : ITagRepository
+    {
+        private readonly string _connectionString;
+        public TagRepository(IConfiguration configuration)
+        {
+            _connectionString = configuration.GetConnectionString("DefaultConnection");
+        }
+
+        private SqlConnection Connection
+        {
+            get { return new SqlConnection(_connectionString); }
+        }
+
+        public List<Tag> GetAll()
+        {
+            using (var conn = Connection)
+            {
+                conn.Open();
+                using (var cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = "SELECT Id, [Name] FROM Tag ORDER BY Name";
+                    using (SqlDataReader reader = cmd.ExecuteReader())
+                    {
+                        var tags = new List<Tag>();
+                        while (reader.Read())
+                        {
+                            tags.Add(new Tag()
+                            {
+                                Id = reader.GetInt32(reader.GetOrdinal("Id")),
```

Comment:
See if you can figure out how to use the DbUtils methods to do these data reads instead of using the reader object directly. Ask your teammates if you're not sure.

Reply:
I figured out DbUtils and got it to work. Glad you pointed that out!